sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -576,6 +576,10 @@ class SessionCatalog(
     tempViews.get(formatTableName(name))
   }
 
+  def getTempViewNames(): Seq[String] = synchronized {
```

Comment:
What about `clearTempTables()` - I wonder if it should also uncache? it's only used for testing now, but could be used in closeSession() to uncache if it also uncached temp views.

Reply:
IIUC, SessionCatalog (HiveSessionCatalog) should be GC when session closed. Manually `clearTempTables` here also fine.

Reply:
> I wonder if it should also uncache? it's only used for testing now.

I am sorry, what is used for testing?

Reply:
The comment on `clearTempTables()` says it is currently only used for testing. If used in `closedSession()` we could remove that.

Reply:
Ah, got it.